hr_attendance_ip_check/models/hr_attendance.py

```diff
@@ -0,0 +1,35 @@
+import logging
+from odoo import models, api, _
+
+_logger = logging.getLogger(__name__)
+
+
+class HrAttendance(models.Model):
+    _inherit = 'hr.attendance'
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        """Override create to validate IP before creating attendance records."""
+        if not vals_list:
+            return {'warning': _('No valid attendance records to create')}
+
+        valid_vals = []
+        for vals in vals_list:
+            employee = self.env['hr.employee'].browse(vals.get('employee_id'))
+            validation_result = employee._validate_ip_address('check_in')
+
+            if isinstance(validation_result, dict):
+                return validation_result
```

Comment:
Warnings cannot be returned from create functions its only used for onchange functions. Instead of warnings throw ValidationError directly from _validate_ip_address function